### PR TITLE
Make tests work with views=true,xattrs=false

### PR DESCRIPTION
**Pull request by torcolvin**
- `user_xattr_key` is only available when xattrs are enabled
- `TestResyncRegenerateSequencesCorruptDocumentSequence` could be made to work without xattrs, but this doesn't feel a worthwhile use of time.

- [x] `GSI=true,xattrs=false` https://jenkins.sgwdev.com/job/SyncGateway-Integration/3076/
- [x] `GSI=true,xattrs=false` https://jenkins.sgwdev.com/job/SyncGateway-Integration/3081/
- [x] `GSI=false,xattrs=false` https://jenkins.sgwdev.com/job/SyncGateway-Integration/3078/ has failures, but they seem to occur only when running a fuller set of tests together and have to do with views service being unhappy
- [x] `GSI=false,xattrs=false` https://jenkins.sgwdev.com/job/SyncGateway-Integration/3079/ has failures, but they seem to occur only when running a fuller set of tests together and have to do with views service being unhappy

### CODE REVIEW

File: rest/persistent_config_test.go [hunk 1]
```diff
@@ -1493,15 +1493,18 @@ func persistentConfigTestCases() []persistentConfigTestCase {
...
+			name:        "document_persistence",
+			xattrConfig: false,
+		},
+	}
+	if base.TestUseXattrs() {
+		testCases = append(testCases, persistentConfigTestCase{
+			name:        "xattr_persistence",
+			xattrConfig: true,
+		})
+	}
```
Comment (bbrks) [inline rest/persistent_config_test.go]: This doesn't seem right and I'm curious if anything broke with this because there's an underlying bug, or whether this was just cleanup?

We should continue testing both unconditionally, because it's valid to set `UseXattrConfig` for SG config even when `"enable_shared_bucket_access": false` (which is what `SG_TEST_USE_XATTRS` is intended for)

File: db/util_testing.go [hunk 2]
```diff
@@ -206,11 +204,7 @@ func purgeWithDCPFeed(ctx context.Context, dataStore sgbucket.DataStore, tbp *ba
 
 		key := string(event.Key)
 
-		if base.TestUseXattrs() {
-			purgeErr = dataStore.DeleteWithXattrs(ctx, key, []string{base.SyncXattrName})
-		} else {
-			purgeErr = dataStore.Delete(key)
-		}
+		purgeErr := dataStore.DeleteWithXattrs(ctx, key, []string{base.SyncXattrName})
```
Comment (bbrks) [inline db/util_testing.go:208]: This confused me at first but does make sense since `DeleteWithXattrs` will handle both xattr and non-xattr states of documents the same way.

Might be worth a comment?

**Review by bbrks — DISMISSED** [review]

File: rest/persistent_config_test.go (hunk 1 above)
Comment (torcolvin) [reply rest/persistent_config_test.go]: I thought this would just make it more obvious what was going on, testing xattr persistence is for capella and non xattr is not that common. This issue is fixed by switching the bucket emptier to `purgeErr := dataStore.DeleteWithXattrs(ctx, key, []string{base.SyncXattrName})` since what happens with xattr persistence tests is that the sync metadata is preserved between tests, and an earlier test is not cleaned up for a later test.

This is actually a bug with xattrs=false,views=true since the views bucket pool uses bucket flush and that's less of a problem.

File: db/util_testing.go (hunk 2 above)
Comment (torcolvin) [reply db/util_testing.go:208]: Added a comment for this.

File: rest/persistent_config_test.go (hunk 1 above)
Comment (torcolvin) [reply rest/persistent_config_test.go]: Readded these tests.

File: rest/adminapitest/admin_api_test.go [hunk 3]
```diff
@@ -2898,9 +2901,9 @@ func TestConfigsIncludeDefaults(t *testing.T) {
 	tb2 := base.GetTestBucket(t)
 	defer tb2.Close(ctx)
 
-	resp = rest.BootstrapAdminRequest(t, sc, http.MethodPut, "/db2/",
-		`{"bucket": "`+tb2.GetName()+`", "num_index_replicas": 0, "unsupported": {"disable_clean_skipped_query": true}}`,
-	)
+	resp = rest.BootstrapAdminRequest(t, sc, http.MethodPut, "/db2/", fmt.Sprintf(
+		`{"bucket": "%s", "num_index_replicas": 0, "num_index_replicas": 0, "enable_shared_bucket_access": %t, "use_views": %t, "unsupported": {"disable_clean_skipped_query": true}}`, tb2.GetName(), base.TestUseXattrs(), base.TestsDisableGSI(),
+	))
```
Comment (gregns1) [inline rest/adminapitest/admin_api_test.go:2906]: Is there a reason for `"num_index_replicas"` repeated here?

File: rest/xattr_upgrade_test.go [hunk 4]
```diff
@@ -224,8 +227,10 @@ func TestCheckForUpgradeFeed(t *testing.T) {
 	assert.NoError(t, err, "Error writing doc w/ xattr")
 	rt.WaitForSequence(1)
 
+	// Use non admin channels since it is in the channel cache. Using admin using * channel which will not find this document.
+
 	// Attempt to update the documents via Sync Gateway.  Should trigger checkForUpgrade handling to detect metadata in xattr, and update normally.
-	changes := rt.PostChangesAdmin("/{{.keyspace}}/_changes", "{}")
+	changes := rt.GetChanges("/{{.keyspace}}/_changes", user)
```
Comment (gregns1) [inline rest/xattr_upgrade_test.go]: I am confused about what this test is testing for here. But changing this to get changes by user means this will just return the user doc. If that is what the test is trying to do (I'm really not sure myself) then we should alter the above comments to make it clear that the documnet we expect to see on changes here is the user doc and not `TestCheckForUpgrade` doc. 

If this test is trying to see the document `TestCheckForUpgrade` added above this changes call doesn't return that I think because the documents metadata doesn't actually have channel A allocated to it `"channels": [null]`.

**Review by gregns1 — COMMENTED** [review]
Mostly looks good just couple of questions I have

File: rest/adminapitest/admin_api_test.go (hunk 3 above)
Comment (torcolvin) [reply rest/adminapitest/admin_api_test.go:2906]: accident, I've added a helper function for this.

Comment (torcolvin) [reply rest/adminapitest/admin_api_test.go:2906]: I didn't add a helper function but I removed this.

File: rest/xattr_upgrade_test.go (hunk 4 above)
Comment (torcolvin) [reply rest/xattr_upgrade_test.go]: Skipped with a comment, as per discussion with Adam. This doesn't work with GSI and that's expected. This test was for a rolling upgrade from non xattrs->xattrs and the non xattr nodes will already have this value in the cache.

This is an edge case that we won't fix so skipped the test.

**Review by gregns1 — APPROVED** [review]
